modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java

```diff
@@ -1306,6 +1347,147 @@ public void testCorrectPartitionRecoveryOnSnapshot() throws InterruptedException
         }
     }
 
+    @Test
+    public void tableRecoveryOnMultipleRestartingNodesTest() throws InterruptedException {
+        int euIdx = 0;
+        int usIdx = 1;
+        int ruIdx = 2;
+        int jpIdx = 3;
+
+        IgniteImpl nodeEu = startNode(euIdx, configurationString(euIdx, "{ region.attribute = \"EU\" }"));
+        IgniteImpl nodeUs = startNode(usIdx, configurationString(usIdx, "{ region.attribute = \"US\" }"));
+
+        startNode(ruIdx, configurationString(ruIdx, "{ region.attribute = \"RU\" }"));
+        startNode(jpIdx, configurationString(jpIdx, "{ region.attribute = \"JP\" }"));
+
+        WatchListenerInhibitor nodeEuInhibitor = WatchListenerInhibitor.metastorageEventsInhibitor(nodeEu);
+
+        String tableName = "TEST";
+        String startFilter = "$[?(@.region == \"EU\" || @.region == \"US\")]";
+        String alteredFilter = "$[?(@.region == \"RU\" || @.region == \"JP\")]";
+
+        try (Session session = nodeUs.sql().createSession()) {
+            session.execute(null, String.format("CREATE ZONE IF NOT EXISTS ZONE_%s WITH REPLICAS=%d, PARTITIONS=%d, "
+                            + "DATA_NODES_FILTER='%s'", tableName, 3, 1, startFilter));
```

Comment:
it's not a tableName but zoneName probably.

Reply:
its used to create zone name, but I will make the separate variable for zone name